unittests/test_asc_editor.py

```diff
@@ -180,12 +190,14 @@ def test_subcircuit_cell_in_lib(self):
         my_edt = spicelib.editor.asc_editor.AscEditor(test_dir + "testcomp2.asc")
         self.assertAlmostEqual(my_edt["U1:R1"].value, 320, msg="Subcircuit Value for U1:R1, float comparison")
         my_edt["R2"].value = 20
-        self.assertAlmostEqual(my_edt["R2"].value, 20, msg="Subcircuit Value for R2, float comparison after edit")
+        self.assertAlmostEqual(my_edt["R2"].value, 20, msg="Value for R2, float comparison after edit")
         
-        # The following test is not working as it should, because the value is not being saved in save_netlist()
-        # It also doesn't log via the logger, but that is another issue.
-        # my_edt["U1:R1"].value = 330
-        # self.assertAlmostEqual(my_edt["U1:R1"].value, 330, msg="Subcircuit Value for U1:R1, float comparison after edit")
+        self.assertTrue(my_edt.get_subcircuit("U1").is_read_only(), "Subcircuit U1 should be readonly")
```

Comment:
Here it is failing too.

Reply:
Meanwhile, I was thinking that maybe the implementation needs to change a bit, so that the read-only property is inherited from the parent circuit. Yes ?

Reply:
the parent circuit may be OK for modifications. It is only the circuits that come from libraries that should be read-only. Maybe indeed circuits that are loaded from a library should also be read-only. But I'm not sure that we load circuits from inside libraries.

Reply:
The reason that the library circuits are no longer flagged as readonly is because the line that set as "True" has disappeared from the code. I'll check.

Reply:
repaired